apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1083,34 +1098,41 @@ queue_item_marshaller(Item) ->
 estimate_size(QItem) ->
     erlang:external_size(QItem).
 
--spec append_queue(id(), index(), replayq:q(), [queue_query()]) -> replayq:q().
-append_queue(Id, Index, Q, Queries) when not is_binary(Q) ->
-    %% we must not append a raw binary because the marshaller will get
-    %% lost.
+-spec append_queue(id(), index(), replayq:q(), [queue_query()]) ->
+    {[queue_query()], replayq:q()}.
+append_queue(Id, Index, Q, Queries) ->
+    %% this assertion is to ensure that we never append a raw binary
+    %% because the marshaller will get lost.
+    false = is_binary(hd(Queries)),
     Q0 = replayq:append(Q, Queries),
-    Q2 =
+    {Overflown, Q2} =
         case replayq:overflow(Q0) of
-            Overflow when Overflow =< 0 ->
-                Q0;
-            Overflow ->
-                PopOpts = #{bytes_limit => Overflow, count_limit => 999999999},
+            OverflownBytes when OverflownBytes =< 0 ->
+                {[], Q0};
+            OverflownBytes ->
+                PopOpts = #{bytes_limit => OverflownBytes, count_limit => 999999999},
                 {Q1, QAckRef, Items2} = replayq:pop(Q0, PopOpts),
                 ok = replayq:ack(Q1, QAckRef),
                 Dropped = length(Items2),
                 emqx_resource_metrics:dropped_queue_full_inc(Id),
-                ?SLOG(error, #{msg => drop_query, reason => queue_full, dropped => Dropped}),
-                Q1
+                ?SLOG(info, #{
```

Comment:
this log level has been changed from error to info.
because there could be too much to log.